public class ArrayIndexOutOfBoundsExceptionExample {
    public static void main(String[] args) {
       
        int [] arr1 = new int[5];
        try {
            System.out.println(arr1[5]); // Attempt to access an element at index 5, which does not exist
        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Caught an ArrayIndexOutOfBoundsException: " + e.getMessage());
        }

       
        int[] arr2 = new int[5];
        try {
            for (int i = 0; i <= arr2.length; i++) {
                System.out.println(arr2[i]); // Attempt to access an element at index 5, which does not exist
            }
        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Caught an ArrayIndexOutOfBoundsException: " + e.getMessage());
        }

     
        int[][] arr3 = new int[2][3];
        try {
            System.out.println(arr3[2][0]); // Attempt to access an element at index 2, which does not exist
        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Caught an ArrayIndexOutOfBoundsException: " + e.getMessage());
        }

       
        int[] arr4 = new int[3];
        printArray(arr4);
    }

    public static void printArray(int[] arr) {
        try {
            System.out.println(arr[5]); // Attempt to access an element at index 5, which does not exist
        } catch (ArrayIndexOutOfBoundsException e) {
            System.out.println("Caught an ArrayIndexOutOfBoundsException: " + e.getMessage());
        }
    }
}
